extensions-core/caffeine-cache/src/test/java/io/druid/client/cache/CaffeineCacheTest.java

```diff
@@ -0,0 +1,494 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.client.cache;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.primitives.Ints;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.name.Names;
+import com.metamx.common.lifecycle.Lifecycle;
+import io.druid.guice.GuiceInjectors;
+import io.druid.guice.JsonConfigProvider;
+import io.druid.guice.JsonConfigurator;
+import io.druid.guice.ManageLifecycle;
+import io.druid.initialization.Initialization;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.UUID;
+
+public class CaffeineCacheTest
+{
+  private static final byte[] HI = "hiiiiiiiiiiiiiiiiiii".getBytes();
+  private static final byte[] HO = "hooooooooooooooooooo".getBytes();
+
+  private CaffeineCache cache;
+  private final CaffeineCacheConfig cacheConfig = new CaffeineCacheConfig()
+  {
+    @Override
+    public boolean isEvictOnClose()
+    {
+      return true;
+    }
+  };
+
+  @Before
+  public void setUp() throws Exception
+  {
+    cache = CaffeineCache.create(cacheConfig);
+  }
+
+  @Test
+  public void testBasicInjection() throws Exception
+  {
+    final CaffeineCacheConfig config = new CaffeineCacheConfig();
+    Injector injector = Initialization.makeInjectorWithModules(
+        GuiceInjectors.makeStartupInjector(), ImmutableList.of(
+            new Module()
```

Comment:
fyi, I think you could use a lambda expression (SAM interface) so it would be `binder -> { ... }`